samples/performance/bindless/bindless.cpp

```diff
@@ -0,0 +1,925 @@
+				}
+			}
+			model.bounding_sphere = BoundingSphere(pts);
+			models.emplace_back(std::move(model));
+		}
+	}
+
+	auto &cmd = device->get_command_pool().request_command_buffer();
+	cmd.begin(VK_COMMAND_BUFFER_USAGE_ONE_TIME_SUBMIT_BIT, VK_NULL_HANDLE);
+	for (auto &&texture : textures)
+	{
+		VkImageSubresourceRange subresource_range = {VK_IMAGE_ASPECT_COLOR_BIT, 0, 1, 0, 1};
+		subresource_range.baseMipLevel            = 0;
+		subresource_range.levelCount              = texture.n_mip_maps;
+
+		VkImageMemoryBarrier image_barrier = vkb::initializers::image_memory_barrier();
+		image_barrier.srcAccessMask        = 0;
+		image_barrier.dstAccessMask        = 0;
+		image_barrier.image                = texture.image->get_handle();
+		image_barrier.subresourceRange     = subresource_range;
+		image_barrier.oldLayout            = VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL;
+		image_barrier.newLayout            = VK_IMAGE_LAYOUT_SHADER_READ_ONLY_OPTIMAL;
+
+		// The semaphore takes care of srcStageMask.
+		vkCmdPipelineBarrier(cmd.get_handle(), VK_PIPELINE_STAGE_TOP_OF_PIPE_BIT, VK_PIPELINE_STAGE_ALL_COMMANDS_BIT, 0, 0, nullptr, 0, nullptr, 1, &image_barrier);
+	}
+	cmd.end();
+	auto &queue = device->get_queue_by_flags(VK_QUEUE_GRAPHICS_BIT, 0);
+	queue.submit(cmd, device->request_fence());
+	device->get_fence_pool().wait();
+}
+
+void BindlessResources::initialize_resources()
+{
+	size_t       vertex_buffer_size = 0, index_buffer_size = 0;
+	const size_t model_buffer_size = models.size() * sizeof(SceneModel);
+	for (auto &&model : models)
+	{
+		model.vertex_buffer_offset = vertex_buffer_size;
+		model.index_buffer_offset  = index_buffer_size;
+
+		vertex_buffer_size += model.vertices.size() * sizeof(Vertex);
+		index_buffer_size += model.triangles.size() * sizeof(model.triangles[0]);
+	}
+
+	assert(vertex_buffer_size && index_buffer_size && model_buffer_size);
+	auto staging_vertex_buffer = std::make_unique<vkb::core::Buffer>(get_device(), vertex_buffer_size, VK_BUFFER_USAGE_TRANSFER_SRC_BIT, VMA_MEMORY_USAGE_CPU_TO_GPU);
+	auto staging_index_buffer  = std::make_unique<vkb::core::Buffer>(get_device(), index_buffer_size, VK_BUFFER_USAGE_TRANSFER_SRC_BIT, VMA_MEMORY_USAGE_CPU_TO_GPU);
+	auto staging_model_buffer  = std::make_unique<vkb::core::Buffer>(get_device(), model_buffer_size, VK_BUFFER_USAGE_TRANSFER_SRC_BIT, VMA_MEMORY_USAGE_CPU_TO_GPU);
+
+	// We will store the GPU commands in the indirect call buffer
+	constexpr auto default_indirect_flags = VK_BUFFER_USAGE_TRANSFER_SRC_BIT | VK_BUFFER_USAGE_STORAGE_BUFFER_BIT | VK_BUFFER_USAGE_INDIRECT_BUFFER_BIT | VK_BUFFER_USAGE_TRANSFER_DST_BIT;
+	auto           indirect_flags         = default_indirect_flags;
+	if (m_supports_buffer_device)
+	{
+		indirect_flags |= VK_BUFFER_USAGE_SHADER_DEVICE_ADDRESS_BIT;
+	}
+	indirect_call_buffer = std::make_unique<vkb::core::Buffer>(get_device(), models.size() * sizeof(VkDrawIndexedIndirectCommand), indirect_flags, VMA_MEMORY_USAGE_GPU_ONLY);
```

Comment:
This buffer is created with `sharingMode VK_SHARING_MODE_EXCLUSIVE`, however from the code it might look like the culling compute dispatches and rendering could run on different queue families. Should this either be `VK_SHARING_MODE_CONCURRENT`, or there needs to be ownership transfer as part of the compute -> graphics handover?